res/controllers/midi-components-0.0.js

```diff
@@ -818,21 +818,20 @@
             this.setCurrentUnit(this.unitNumbers[index]);
         };
 
-        if (unitNumbers !== undefined) {
```

Comment:
Why did you remove this check?

Reply:
Because it's superfluous and insufficient. The warning should be printed if `unitNumbers` is neither an array or a number. This is the case now. Before, the warning was only printed if it was a `undefined`, but not if it was `null` or an object.

Reply:
The problem is that QScriptEngine stops executing silently when trying to do anything with an undefined variable besides check it is undefined.

Reply:
Really? That sounds like an absolutely critical bug... Is that fixed with QJSEngine?

Reply:
This bug has always been there...